modules/standard/IO.chpl

```diff
@@ -2599,14 +2677,32 @@ This function is equivalent to calling :proc:`open` and then
 proc openreader(path:string,
                 param kind=iokind.dynamic, param locking=true,
                 start:int(64) = 0, end:int(64) = max(int(64)),
-                hints:iohints = IOHINT_NONE,
-                style:iostyle = defaultIOStyle())
+                hints:iohints = IOHINT_NONE)
     : channel(false, kind, locking) throws {
+  return openreaderHelper(path, kind, locking, start, end, hints);
+}
+
+private proc openreaderHelper(path:string,
+                              param kind=iokind.dynamic, param locking=true,
+                              start:int(64) = 0, end:int(64) = max(int(64)),
+                              hints:iohints = IOHINT_NONE,
+                              style:iostyleInternal = defaultIOStyleInternal())
+  : channel(false, kind, locking) throws {
 
   var fl:file = try open(path, iomode.r);
-  return try fl.reader(kind, locking, start, end, hints, style);
+  return try fl.readerHelper(kind, locking, start, end, hints, style);
 }
 
+deprecated "openwriter with a style argument is deprecated, please rely on the default value for the argument"
```

Comment:
Should all of these deprecated functions be last resort? I see it on some but not others.